wolfSSL/src/chapter02.md

```diff
@@ -394,301 +394,1177 @@ Removes code for OAEP padding.
+#### WOLFSSL_SESSION_TIMEOUT
 
-#### WOLFSSL_STATIC_PSK
+Default session resumption cache timeout in seconds is used to define timeout manually.
 
-Static ciphers are highly discouraged see [`WOLFSSL_STATIC_RSA`](#wolfssl_static_rsa)
+#### KEEP_OUR_CERT
 
-#### WOLFSSL_STATIC_DH
+Used to ensure the ability to return SSL certificate.
 
-Static ciphers are highly discouraged see [`WOLFSSL_STATIC_RSA`](#wolfssl_static_rsa)
+#### KEEP_PEER_CERT
 
-#### HAVE_NULL_CIPHER
+Retains peer certs. Parts of the OpenSSL compatibility layer require peer certs.
 
-Turns on support for NULL ciphers. This option is highly discouraged from a security standpoint however some systems are too small to perform encrypt/decrypt operations and it is better to at least authenticate messages and peers to prevent message tampering than nothing at all!
+#### WOLFSSL_SIGNER_DER_CERT
 
-#### HAVE_ANON
+Used to no CERT DER buffer OR NULL cm. Used to allocate DER and cert.
 
-Turns on support for anonymous cipher suites. (Never recommended, some valid use cases involving closed or private networks detached from the web)
+#### CA_TABLE_SIZE
 
-#### HAVE_LIBOQS
+wolfSSL certificate manager. CA signer table. Used in certificate manager and be defined size.
 
-Turn on support for the OpenQuantumSafe team's liboqs integration. Please see the appendix "Experimenting with Quantum-Safe Cryptography" in this document for more details.
+#### ECDHE_SIZE
 
-#### WOLFSSL_SP_4096
+Define to allow this to be overridden at compile-time. ECDHE server size defaults to 256 bits, which can set a predetermined ECDHE curve size. The default is 32 bytes.
 
-Enable RSA/DH 4096-bit Single-Precision (SP) support.
+#### CIPHER_NONCE
 
-#### WOLFSSL_SP_384
+It is used as a cryptographic number, which is implemented in authentication. It is a pseudo-random number, which is an integrity-only cipher suite.
 
-Enable ECC SECP384R1 Single-Precision (SP) support. Only applies to `WOLFSSL_SP_MATH`.
+#### WOLFSSL_USE_POPEN_HOST
 
-#### WOLFSSL_SP_1024
+Used to implement popen and "host".
+
+#### CloseSocket
+
+Cleanup of the socket close code used, for example, CRL/OCSP and BIO, to use single macro `CloseSocket.`
+
+#### CONFIG_POSIX_API
+
+We need POSIX names for networking systems calls. This can either be enabled with CONFIG_NET_SOCKETS_POSIX_NAMES or with CONFIG_POSIX_API. This commit enables support for the latter."
+
+#### WOLFSSL_USER_CURRTIME
+
+Add in the option to use in test.h without the gettimeofday function using the macro WOLFSSL_USER_CURRTIME.
+
+#### WOLFSSL_USER_MUTEX
+
+Option for user-defined mutexes with WOLFSSL_USER_MUTEX.
+
+#### DEFAULT_MIN_ECCKEY_BITS
+
+Identifies the minimum number of bits in ECCkey.
+
+#### DEFAULT_MIN_RSAKEY_BITS
+
+Identifies the minimum number of bits in RSA key.
+
+#### EXIT_FAILURE
```

Comment:
This is actually defined by the system normally (`stdlib.h`) and can't be overridden.  Probably doesn't make sense to document here.